velox/functions/prestosql/tests/Utf8Test.cpp

```diff
@@ -15,7 +15,7 @@
  */
 
 #include <gtest/gtest.h>
-#include "velox/functions/prestosql/Utf8Utils.h"
```

Comment:
Is this a test for Utf8Utils that were moved to lib? If so, the test needs to move to lib/tests.

The logic in Utf8Utils is Presto-specific. Does it happen to match Spark's behavior? Would you clarify? CC: @rui-mo

Reply:
Spark doesn't handle for invalid UTF-8 and wide-character,  spark only guarantee work well for char with size at most 2 bytes. It should be fine here to leverage Utf8Utils.

Reply:
> spark only guarantee work well for char with size at most 2 bytes 

UDF-8 utilities we have support characters up to 4 bytes long. Are you saying Spark only allows 2-byte characters? If so, do we expect different behavior for strings with 3-4 byte characters? CC: @rui-mo

Reply:
sorry @mbasmanova , i mix other string function with this split function. Dobule checked for split function, spark convert it to `UTF8String` class to do split action, it can handle `wide character` well,  looks the only case spark can't handle well is the case if input string has `invalid UTF-8` and delimiter is empty.

![image](https://github.com/user-attachments/assets/2e4cb38e-f6f4-4bed-bbec-cd3c087f8c34)

Checking code here - https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala#L581

Reply:
Looks like inputs are assumed to be valid UTF8 characters as @mbasmanova  mentioned at https://github.com/facebookincubator/velox/pull/10264#discussion_r1684218012.

Reply:
@mbasmanova

Reply:
@rui-mo @gaoyangxiaozhu @PHILO-HE To avoid confusion going forward, should we update documentation in https://facebookincubator.github.io/velox/functions/spark/string.html to clearly state that input strings are expected to be vaild UTF-8 and that behavior is undefined if they are not?

Reply:
Opened https://github.com/facebookincubator/velox/pull/10682. Thanks.